src/xenia/vfs/devices/stfs_container_device.cc

```diff
@@ -182,23 +184,34 @@ Entry* StfsContainerDevice::ResolvePath(const std::string_view path) {
 }
 
 StfsContainerDevice::Error StfsContainerDevice::ReadHeaderAndVerify(
-    const uint8_t* map_ptr, size_t map_size) {
-  // Copy header & check signature
-  memcpy(&header_, map_ptr, sizeof(StfsHeader));
-  if (header_.header.magic != XContentPackageType::kPackageTypeCon &&
-      header_.header.magic != XContentPackageType::kPackageTypeLive &&
-      header_.header.magic != XContentPackageType::kPackageTypePirs) {
+    FILE* header_file) {
+  // Check size of the file is enough to store an STFS header
+  xe::filesystem::Seek(header_file, 0L, SEEK_END);
+  files_total_size_ = xe::filesystem::Tell(header_file);
+  xe::filesystem::Seek(header_file, 0L, SEEK_SET);
+
+  if (sizeof(StfsHeader) > files_total_size_) {
+    return Error::kErrorTooSmall;
+  }
+
+  // Read header & check signature
+  fread(&header_, sizeof(StfsHeader), 1, header_file);
```

Comment:
`fread`s (not just this one) could probably use return value checking to ensure the count matches what we expect.